src/lib/types/Product.ts

```diff
@@ -18,6 +18,10 @@ export interface ProductTranslatableFields {
 		href: string;
 		fallback?: boolean;
 	}[];
+	variations?: {
+		name: string;
+		value: string;
+	}[];
```

Comment:
This can't be stored in `ProductTranslatableFields` in my opinion. This determines how the backend works (at checkout, ...) and also whether or not the product needs to be standalone.

This structure is more for frontend fields.

Maybe something like this instead:

```suggestion
	variationLabels?: {
		names: Record<string,string>; // name by category id
		values: Record<string, Record<string, string>>; // value by value id inside category id  
	}[];
```

With a `variations` field instead inside `Product.ts` - that will contain ids - and later prices. The ids can be internal/hidden (eg set initially when creating the variation with `generateId` function, and then reused smartly by the frontend)

Reply:
Can you explain the content of each Record ? An example ?

Reply:
For example, two variations: "Color" "Blue" and "Color" "Red"

It's stored as : 

```ts
  variations: [{name: "color", value: "blue"}, {name: "color", value: "red"}],
  variationLabels: {values: {color: {red: "Red", blue: "Blue"}}, names: {color: "Color"}},
  translations: {
    fr: { variationLabels: {values: {color: {red: "Rouge", blue: "Bleu"}}, names: {color: "Couleur"}}}
  }
```